torchmetrics/text/eed.py

```diff
@@ -0,0 +1,87 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Optional, Sequence, Union
+
+from torch import Tensor, tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.text.eed import _eed_compute, _eed_update
+from torchmetrics.metric import Metric
+
+
+class EED(Metric):
+    """Computes extended edit distance score (`EED`_) [1] for strings or list of strings The metric utilises the
+    Levenshtein distance and extends it by adding an additional jump operation.
+
+    Args:
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Returns:
+        Extended edit distance score as a tensor
+
+    Example:
+        >>> hypotheses = ["this is the prediction", "here is an other sample"]
+        >>> references = ["this is the reference", "here is another one"]
+        >>> metric = EED()
+        >>> metric(hypothesis_corpus=hypotheses, reference_corpus=references)
+        tensor(0.3204)
+
+    References:
+        [1] P. Stanchev, W. Wang, and H. Ney, “EED: Extended Edit Distance Measure for Machine Translation”, submitted
+        to WMT 2019. `EED`_
+    """
+
+    scores: Tensor
+    total: Tensor
+
+    def __init__(
+        self,
+        language: Literal["en", "ja"] = "en",
+        compute_on_step: bool = True,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None,
+    ):
+        super().__init__()
+        self.language: Literal["en", "ja"] = language
+        self.add_state("scores", tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("total", tensor(0.0), dist_reduce_fx="sum")
+
+    def update(
+        self,
+        hypothesis_corpus: Union[str, Sequence[str]],
+        reference_corpus: Sequence[Union[str, Sequence[str]]],
```

Comment:
As for other NMT metrics, I would propose to have `references` and the `hypothesis` in this order to keep this consistent and allow a user to use `MetricCollection` smoothly.

Reply:
agree, changed code for next commit